File: python/entry.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import sys

def main():
	if len(sys.argv) < 2:
		print("Please enter an integer value")
		sys.exit()

	n = int(sys.argv[1])
	print("Fact(" + str(n) + ")= " + str(fact(n)))
	print sys.argv[0]

def fact(i):
	if i <= 0:
		return 1
	return i * fact(i-1)

if __name__ == "__main__":
	main()

File: codes/cpp/hello/entry.cpp
#include <iostream>
#include <cstdlib>

int fact(int i);

using namespace std;
int main(int argc, char *argv[])
{

  if (argc < 2) {
    cout << "Please enter an integer value\n";
    return 1;
  }

  int n = atoi(argv[1]);
  cout << "Fact(" << n << ")= " << fact(n) << "\n" ;
}

int fact(int i){
  if (i <=0) return 1;
  return i * fact(i-1);
}
